cmd/snap-bootstrap/bootstrap/bootstrap.go

```diff
@@ -152,15 +160,61 @@ func Run(gadgetRoot, device string, options Options) error {
 	// store the encryption key as the last part of the process to reduce the
 	// possiblity of exiting with an error after the TPM provisioning
 	if options.Encrypt {
-		if err := rkey.Store(options.RecoveryKeyFile); err != nil {
-			return fmt.Errorf("cannot store recovery key: %v", err)
+		if err := tpmSealKey(key, rkey, options); err != nil {
+			return fmt.Errorf("cannot seal the encryption key: %v", err)
 		}
+	}
+
+	return nil
+}
+
+// TODO:UC20: get cmdline definition from bootloaders
+var kernelCmdlines = []string{
+	// run mode
+	"console=ttyS0 console=tty1 panic=-1 systemd.gpt_auto=0 init=/sbin/init snapd_recovery_mode=run",
```

Comment:
this is wrong now, we need to boot with:

```suggestion
	"console=ttyS0 console=tty1 panic=-1 systemd.gpt_auto=0 rd.systemd.unit=basic.target snapd_recovery_mode=run",
```

see https://github.com/snapcore/pc-amd64-gadget/commit/9d476ec1069b38884e3d7595e8904ecce2b8eef9 for the pc gadget, and there's an equivalent change in the pi gadget too

Reply:
Oh oops, I think it's time to update my gadgets here.

Reply:
@xnox tells me that the new gadget files reverted shim to an older version so we'll also need this fixed in order to be able to unseal again.